schemasheets/template_wizard.py

```diff
@@ -0,0 +1,1011 @@
+import csv
+import logging
+import os
+import pprint
+import re
+from dataclasses import dataclass
+from typing import List, Dict, Optional, Any
+
+import click
+import click_log
+import yaml
+from linkml_runtime import SchemaView
+from linkml_runtime.dumpers import yaml_dumper
+from linkml_runtime.linkml_model import EnumDefinition, PermissibleValue
+
+from schemasheets.schema_exporter import SchemaExporter
+
+logger = logging.getLogger(__name__)
+click_log.basic_config(logger)
+
+SLOT_NAME = str
+METAMODEL_SLOT_NAME = str
+PROJECT_SLOT_NAME = str
+SLOT_VALUE = str
+SLOT_USAGE_COUNT = int
+
+
+# todo test first
+
+# todo too much writing to file and reading back in
+#  also, there must be something better than LinkML->YAML->Dict
+
+# todo: reflect the classes that use a given slot on slot-only rows (via domain of)
+#  see prepare_merged_report() workarround
+
+# todo: work on YAML-serialized alt_descriptions columns
+
+# todo: write a style guide, esp use of external terms and consistent grammar
+#  NMDC Schema Style Guide
+#  https://docs.google.com/document/d/1KwO5kG79MWuS4eaRn6n9B3Tth4Hg_vo7jPQDqdv-yD4/edit#
+
+# todo: group and or describe mappings
+# todo: comments, notes, todos distinctions
+
+# todo: clearer class and method names, docstrings, typing
+
+
+@dataclass
+class TemplateWizard:
+    """
+    models the intermediate states while converting the metamodel and some project schema into a bare,
+    then populated schemsheets template
+    """
+
+    meta_view: SchemaView
+    project_view: SchemaView
+
+    slot_attributes: Optional[
+        Dict[SLOT_NAME, Dict[METAMODEL_SLOT_NAME, SLOT_VALUE]]
+    ] = None
+
+    # todo refactor this? dict with name and count had been handy for writing to CSV
+    #  but were not doing that anymore
+    proj_elem_usage: Optional[List[Dict[PROJECT_SLOT_NAME, Any]]] = None
+
+    class_slots_slotnames: Optional[List[PROJECT_SLOT_NAME]] = None
+
+    def get_element_annotations(self) -> List:
+        """
+        iteratively collects all element annotations
+        """
+        elements = self.project_view.all_elements()
+        annotations = set()
+        for ek, ev in elements.items():
+            if ev.annotations:
```

Comment:
minor: unnecessary, the way pythongen works (and meta.py is made by pythongen) is such that every list
is prepopulated as an empty list